roles/edpm_nova/tasks/configure.yml

```diff
@@ -137,3 +137,40 @@
     owner: nova
     group: nova
   when: edpm_nova_tls_certs_enabled
+
+- name: Check if compute_id exists in state_path
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.stat:
+    path: /var/lib/nova/compute_id
+  register: compute_id
+
+- name: Pre-generate compute_id if not already exists
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.copy:
+    dest: "/var/lib/nova/compute_id"
+    owner: nova
+    group: nova
+    mode: "0400"
+    attributes: "+i"
+    content: |
+      {{ lookup('ansible.builtin.pipe', 'uuidgen') }}
+  when: not compute_id.stat.exists
+
+- name: Make compute_id ready-only
```

Comment:
is there a reason this task is needed given the previous task also sets all the same file properties?

Reply:
this extends the above lines for cases when this file is already exists and is not being created

Reply:
oh ok...i would suggest adding that "when".

Reply:
Done.